streams/lms_metadata.py

```diff
@@ -113,29 +119,26 @@ def connect(self):
             meta["track"] = song_data["title"]
             meta["image_url"] = song_data["artwork_url"]
         # File locking so to reduce errors, without locks here and on the read cycle you can sometimes read while writing, which will read an empty file and crash the stream
-        f = open(f"lms_{str(self.player_name).replace(' ', '_')}_metadata.json", 'wt', encoding='utf-8')
-        try:
-          fcntl.flock(f, fcntl.LOCK_EX)
-          json.dump(meta, f, indent = 2)
-        finally:
-          fcntl.flock(f, fcntl.LOCK_UN)
-          f.close()
+        with open(f"lms_{str(self.player_name).replace(' ', '_')}_metadata.json", 'wt', encoding='utf-8') as f:
+          try:
+            fcntl.flock(f, fcntl.LOCK_EX)
+            json.dump(meta, f, indent = 2)
+          finally:
+            fcntl.flock(f, fcntl.LOCK_UN)
 
         if self.dump:
-          f = open(f"{str(self.player_name).replace(' ', '_')}_track_raw.json", "w")
-          json.dump(track_load, f, indent = 2)
-          f.close()
-          f = open(f"{str(self.player_name).replace(' ', '_')}_song_raw.json", "w")
-          json.dump(song_load, f, indent = 2)
-          f.close()
+          with open(f"{str(self.player_name).replace(' ', '_')}_track_raw.json", "w") as f:
+            json.dump(track_load, f, indent = 2)
+          with open(f"{str(self.player_name).replace(' ', '_')}_song_raw.json", "w") as f:
+            json.dump(song_load, f, indent = 2)
       except Exception as e:
         print(f"Error: {e}, trying again in {self.meta_ref_rate} seconds...")
 
       # a sleep equal to the meta_ref_rate, that way the metadata refreshes on a set schedule while looping instead of just doing it at all times always
       time.sleep(self.meta_ref_rate)
 
 if __name__ == '__main__':
-  parser = argparse.ArgumentParser(description='LMS Metadata')
+  parser = argparse.ArgumentParser(description="LMS Metadata Reader - a script for finding an LMS player with a given name and extracting the name of the song, album, and artist as well as getting the album picture")
   parser.add_argument('--name', type=str, required=True, help='The name of the LMS Player')
   parser.add_argument('--ref', type=int, default=2, help='The frequency of metadata refresh cycles')
   parser.add_argument('--dump', action='store_true', help="""Create raw json dumps directly from the LMS server,
```

Comment:
maybe call this debug and use it to enable some of the extra printouts as well/?